docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +164,104 @@ Note that:
+
+
+**Details to Aspect 2: Consuming/providing configuration files via network**
+
+For aspect (2) there are two proposals as below. Decision has to been taken which proposal to follow.
+
+**Proposal 1:**
+
+The `c8y_configuration_plugin` configuration's record field `path` (reference to section 'Configuration' above) can be prefixed with "mqtt://". Then it's value is treated as MQTT topic structure, where another process/external device is expected to provide/consume the configuration file.
+
+The `c8y_configuration_plugin` expects the process/external device always putting the latest config file as retain message to the MQTT topic `{path}`. Then the plugin consumes that retain message whenever a cloud request comes in.
+
+Example Plugin Config:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { path = '/etc/tedge/tedge.toml', type = 'tedge.toml' },
+    { path = 'mqtt://configs/bar.conf', type = 'bar.conf', childid = 'child1'  }
+  ]
+```
+
+Example Flow:
+
+**Start Behaviour:**
+  * external device `child1`: starts
+  * external device `child1`: publishs all its config files to MQTT (with retain); Example: Publish to topic `configs/bar.conf`
+
+**Device-to-Cloud Behaviour:**
+  * at some point a config retrieval for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: subscribes to `configs/bar.conf` and gets immediately the config file content from the retained message
+  * C8Y config plugin: sends the recived MQTT payload as config file to C8Y<br/><br/>
+    NOTE: The responsibility to assure that the latest config file content is on the MQTT bus is always on the process/external device.
+When there is no retain message the plugin sends an error to the cloud on an every incoming config retrieval request.
+
+**Cloud-to-Device Behaviour:**
+  * at some point a config sent from cloud for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: publishs the config file content as MQTT retain message to `configs/bar.conf` 
+  * external device `child1`: recognizes the MQTT message on `configs/bar.conf` and processes the new config file content
+
+**Proposal 2:**
+
+The `c8y_configuration_plugin` provides the field `protocol` for all records in the `c8y_configuration_plugin` configuration (reference to section 'Configuration' above). If `protocol` is set to "http" the `c8y_configuration_plugin` expects another process/external device to provide/consume the configuration file via HTTP protocol. Thereby the records field `path` is used with the prefix `http://<thin-edge device IP address>/tedge/configurations/` as URL. Example: `http://192.168.1.6/tedge/configurations/child1/bar.conf` where (192.168.1.6 is the IP address of the thin-edge device)
+
+The `c8y_configuration_plugin` expects the process/external device always putting the latest config file to the thin-edge device. Then the plugin consumes that file whenever a cloud request comes in.<br/> 
```

Comment:
This is a general assumption whatever the protocol used.

One can also add an MQTT request for the child device to upload its current configuration.

Reply:
As we introduced HTTP filetransfer supported by `tedge_agent`, the plugin has no more to deal with a protocol in direction to the external device, consequently and field `protocol` was removed.